src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -118,6 +120,44 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
       else Iterator.empty.next()
   }
 
+  /*
+   * Implemented by means of a buffer to keep track of the last n elements during iteration.
+   */
+  def takeRight(n: Int): Iterator[A] = {
+    if (n == 0) Iterator.empty
+    else {
+      // Return an iterator that iterates over the elements via a buffer
+      new Iterator[A]() {
+        private[this] var index = 0
```

Comment:
I’m curious about the `private[this]`: since we are in an anonymous class, everything is going to be private, right? Or maybe it has an impact on optimizations?